src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -758,7 +759,7 @@ def __init__(self, *args, **kwargs):
         ])
 
         if not ca_signed_certs:
-            return (await self._get_instance(ca_id))['serial'] + 1
+            return ctypes.c_ulong((await self._get_instance(ca_id))['serial']) + 1
```

Comment:
Why ctypes here?

Reply:
I still dont get what ctypes has to do with any of this, can you explain, please?

Reply:
This is just casting the serial to an unsigned 64-bit integer

Reply:
@jhixson74 as far as I know thats converting an integer to a ctypes object, which is not serializable and will break the API.

Python does not have integer size, its just integer, it will accommodate automatically to any size

>>> 2 ** 160
1461501637330902918203684832716283019655932542976
>>> int('1461501637330902918203684832716283019655932542976')
1461501637330902918203684832716283019655932542976

Reply:
Fixed.